bin/strings.swift

```diff
@@ -55,69 +55,71 @@ func flatten(data: [String:AnyObject], prefix: String = "") -> [String:String] {
   }
 }
 
-func stringsFileContents(strings: [String:String]) -> String {
-  return strings.keys.sort()
+func stringsFileContents(_ strings: [String:String]) -> String {
+  return strings.keys.sorted()
     .filter { key in !key.hasSuffix(".") }
-    .map { key in "\"\(key)\" = \"\(escaped(string: strings[key]!))\";" }
-    .joinWithSeparator("\n")
+    .map { key in "\"\(key)\" = \"\(escaped(strings[key]!))\";" }
+    .joined(separator: "\n")
 }
 
-func funcArgumentNames(string: String) -> [String] {
+func funcArgumentNames(_ string: String) -> [String] {
   return string
-    .componentsSeparatedByString("%{")
-    .flatMap { $0.componentsSeparatedByString("}") }
-    .enumerate()
+    .components(separatedBy: "%{")
+    .flatMap { $0.components(separatedBy: "}") }
+    .enumerated()
     .filter { idx, _ in idx % 2 == 1 }
     .map { _, x in x }
     .distincts(==)
 }
 
-func funcArguments(argumentNames: [String], count: Bool) -> String {
+func funcArguments(_ argumentNames: [String], count: Bool) -> String {
   return argumentNames
-    .map { x in
+    .enumerated()
```

Comment:
oh good catch! that was left over from when we used to have to repeat the first named arg for swift 2, but no longer necessary for swift 3! 

will update!